### PR TITLE
chore: update blueprints to 2.0.0-alpha-47

### LINKED ISSUE #78: Fix generators
Currently almost all generators are breaking. That's for two reasons:

1. Karma integration is not ready to support deep paths (e.g.: `app/services/foo.ts`)
2. Test code is broken

### LINKED ISSUE #84: upgrade blueprints to 2.0.0-alpha.47
- [x] package.json needs to be updated (we intentionally bump the version manually each time for now to have control over breaking changes that require blueprints update)
- [x] replace `import ... from 'angular2/angular2'` with `... from 'angular2/core'` (this might still change in the future, see https://github.com/angular/angular/issues/5539, but for now let's switch to `core`)
- [x] the only exception to the above is `bootstrap` which should come from `import {bootstrap} from 'angular2/platform/browser'`

that should be it...

cc: @wardbell for documentation update since that will be affected in similar way

### PR DESCRIPTION
**Pull request by cironunes**
To do:
- [x] Fix `karma-config-shim` to handle deep folder structures
- [x] Fix service generator
- [x] Fix pipe generator
- [x] Fix component generator

Closes #84
Closes #78

### CODE REVIEW

File: addon/ng2/blueprints/ng2/files/karma-test-shim.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ System.config({
   }
 });
 
-System.import('angular2/src/core/dom/browser_adapter').then(function(browser_adapter) {
+System.import('angular2/src/platform/browser/browser_adapter').then(function(browser_adapter) {
```
Comment (cironunes) [inline addon/ng2/blueprints/ng2/files/karma-test-shim.js]: tried to use just `angular2/platform/browser/browser_adapter` but it didn't work.

Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/karma-test-shim.js]: use `angular2/platform/browser` instead

**Comment by cironunes** [comment]
@IgorMinar I've made all the requested changes, but it seems that TypeScript is not happy when we change from `angular2/angular2` to `angular2/core`. I didn't investigate yet, do you have any insight?

**Comment by IgorMinar** [comment]
I'll check out your branch today and will post back

File: addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {Pipe} from 'angular2/angular2';
 export class <%= classifiedModuleName %> {
 
   transform(value, args?) {
-    return value;
+    return !value;
```
Comment (IgorMinar) [inline addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.ts]: huh?

File: addon/ng2/blueprints/ng2/files/src/app.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import {bootstrap} from 'angular2/angular2';
+import {bootstrap} from 'angular2/platform/browser';
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/app.ts]: above this line add:

```
/// <reference path="node_modules/angular2/manual_typings/globals-es6.d.ts"/>
```

this will resolve all the `tsc` issues - this is a bug in angular a47. I'll file an issue.

let's add this line and we'll remove it once the bug is fixed.

File: addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.spec.ts [hunk 4]
```diff
@@ -15,10 +15,10 @@ import {<%= classifiedModuleName %>} from './<%= dasherizedModuleName %>';
 
 describe('<%= classifiedModuleName %> Pipe', () => {
 
-  beforeEachProviders(() => []);
+  beforeEachProviders(() => [<%= classifiedModuleName%>]);
```
Comment (IgorMinar) [inline addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.spec.ts]: why this change? this looks wrong

File: addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.ts (hunk 2 above)
Comment (IgorMinar) [inline addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.ts]: oh I see. you are trying to make the test pass. how about returning null and updating the test to that too

File: addon/ng2/blueprints/service/files/src/app/services/__name__/__name__.spec.ts [hunk 5]
```diff
@@ -15,11 +15,11 @@ import {<%= classifiedModuleName %>} from './<%= dasherizedModuleName %>';
 
 describe('<%= classifiedModuleName %> Service', () => {
 
-  beforeEachProviders(() => []);
+  beforeEachProviders(() => [<%= classifiedModuleName %>]);
```
Comment (IgorMinar) [inline addon/ng2/blueprints/service/files/src/app/services/__name__/__name__.spec.ts]: ok, here it makes sense

File: addon/ng2/blueprints/component-test/files/src/app/components/__name__/__name__.spec.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ import {<%= classifiedModuleName %>} from './<%= dasherizedModuleName %>';
 
 describe('<%= classifiedModuleName %> Component', () => {
 
-  beforeEachProviders(() => []);
+  beforeEachProviders(() => [<%= classifiedModuleName %>]);
```
Comment (IgorMinar) [inline addon/ng2/blueprints/component-test/files/src/app/components/__name__/__name__.spec.ts]: why this change? this looks wrong

File: addon/ng2/blueprints/service/files/src/app/services/__name__/__name__.ts [hunk 7]
```diff
@@ -6,4 +6,8 @@ export class <%= classifiedModuleName %> {
 
   constructor() {}
 
-}
\ No newline at end of file
+  doSomething() {
```
Comment (IgorMinar) [inline addon/ng2/blueprints/service/files/src/app/services/__name__/__name__.ts]: no, we shouldn't be scaffolding stuff like this. just leave the service empty for now please

File: addon/ng2/blueprints/ng2/files/src/app.ts (hunk 3 above)
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/app.ts]: also add a link to https://github.com/angular/angular/issues/5596 for more context as to why this line is needed

File: addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.spec.ts (hunk 4 above)
Comment (cironunes) [inline addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.spec.ts]: if we don't have the providers injected there it throws:

```
	Failed: No provider for "name"
	Error: DI Exception
```

Comment (cironunes) [inline addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.spec.ts]: please confirm this. other changes were made.

Comment (IgorMinar) [inline addon/ng2/blueprints/pipe/files/src/app/pipes/__name__/__name__.spec.ts]: Oh. Never mind. I was wrong about this one

File: addon/ng2/blueprints/component-test/files/src/app/components/__name__/__name__.spec.ts (hunk 6 above)
Comment (IgorMinar) [inline addon/ng2/blueprints/component-test/files/src/app/components/__name__/__name__.spec.ts]: This one should not be necessary even if it is necessary in pipes blueprint. Can you remove?

**Comment by IgorMinar** [comment]
Otherwise lgtm
